### PR TITLE
Update dependencies and improve build size

### PR DESCRIPTION
**Pull request by daniseijo**
**Summary**

Hi, I am trying to use oneflow at work and I found this library but I wanted to add some features and push them to your repo. But before any change, I have updated most of the dependencies and tried to make some typescript's improvements based on my work with the latest language versions.

I tried to leave everything as it was working before but be sure to check it out and comment any aspect that you think I might have broken.

### CODE REVIEW

**Comment by nikostoulas** [comment]
Wow, thanks for the pr @daniseijo . 
I will have time to check it more mid next week.

File: tsconfig.build.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (nikostoulas) [inline tsconfig.build.json]: What's the use of a second tsconfig? Why not add the include in tsconfig.json

File: bin/oneflow [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env node
 
-require('../build/lib');
+require('../build/src/lib');
```
Comment (nikostoulas) [inline bin/oneflow]: hm, what caused this to change? The tsconfig updates?

File: tsconfig.build.json (hunk 1 above)
Comment (daniseijo) [reply tsconfig.build.json]: You are right, I was having problems with tslint but I have made some test now and I think we don't need this file.

File: bin/oneflow (hunk 2 above)
Comment (daniseijo) [reply bin/oneflow]: This is due to the `resolveJsonModule` change introduced on typescript 3.0. Now `tsc` knows how to adapt the reference to json files on the build folder but it has to have the same depth as in the original root folder. You can check this by making a build and looking at the json files and their paths.

File: tsconfig.build.json (hunk 1 above)
Comment (daniseijo) [reply tsconfig.build.json]: I've deleted the file in the last commit

File: bin/oneflow (hunk 2 above)
Comment (nikostoulas) [reply bin/oneflow]: Thanks for your reply.

**Comment by nikostoulas** [comment]
Thanks for the pr @daniseijo . All tests pass so I am merging it.

**Pull request merged**